### PR TITLE
Key rotation solution

### PR DESCRIPTION
**Pull request by abhaydaga**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/CredRotation/CredRotation.ps1 [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+
+        [Parameter(Mandatory = $true, ParameterSetName = "Other", Position=1, HelpMessage = "Provide the credential name")]
+        #[Parameter(Mandatory = $true, ParameterSetName = "AppService", HelpMessage = "Provide the credential name")]
+        #[Parameter(Mandatory = $true, ParameterSetName = "KeyVault", HelpMessage = "Provide the credential name")]
+        [string]
+		[Alias("cn")]
+        $CredentialName,
+
+        [Parameter(Mandatory = $true, ParameterSetName = "Other", HelpMessage = "Provide the credential location")]
+        #[Parameter(Mandatory = $true, ParameterSetName = "AppService", HelpMessage = "Provide the credential location")]
```
Comment (v-zhgup) [inline src/AzSK/CredRotation/CredRotation.ps1]: Remove commented code

File: src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1 [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+		{
+			$StorageAccount = Get-AzStorageAccount -ResourceGroupName $this.AzSKResourceGroup.ResourceGroupName | Where-Object {$_.StorageAccountName -like 'azsk*'} -ErrorAction SilentlyContinue
+			#if no storage account found then it assumes that there is no control state feature is not used and if there are more than one storage account found it assumes the same
+			$this.AzSKStorageAccount = $StorageAccount;
+		}
+	}
+
+	hidden [PSObject] GetAzSKRG()
+	{
+		$azSKConfigData = [ConfigurationManager]::GetAzSKConfigData()
```
Comment (v-zhgup) [inline src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1]: We have methods in base class to get storage and contexts. Can we use them?

File: src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1 [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+							else{
+								$hash[$setting.Name] = $setting.Value
+							}
+						}
+					}
+					elseif($credentialInfo.appConfigType -eq "Connection Strings"){
+						$this.PublishCustomMessage("Updating the connection string [$($credentialInfo.appConfigName)] in the app service [$($credentialInfo.resourceName)]", [MessageType]::Default) 
+						$appConfig = $resource.SiteConfig.ConnectionStrings
+						$appConfigValue = Read-Host "Enter the new secret for the connection string - [$($credentialInfo.appConfigName)]" -AsSecureString
+						$appConfigValue = [System.Runtime.InteropServices.marshal]::PtrToStringAuto([System.Runtime.InteropServices.marshal]::SecureStringToBSTR($appConfigValue))
```
Comment (v-zhgup) [inline src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1]: Did you test it on linux?

Comment (abhaydaga) [reply src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1]: nope.

File: src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1 (hunk 2 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1]: used the same flow here as in the Partial scan manager.

File: src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1 (hunk 3 above)
Comment (abhaydaga) [reply src/AzSK/Framework/Core/CredRotation/CredRotationScript.ps1]: verified in PSCore. works fine.

**Pull request merged**